### PR TITLE
New BlockaidBannerAlert component

### LINKED ISSUE #19257: Metamask Extension - make blockaid error / warning component configurable
## Background
As part of the [task](https://app.zenhub.com/workspaces/confirmations-secure-ux-6245e6e2348677001213b8d2/issues/gh/metamask/metamask-planning/558) React component will be developed to show errors / warnings returned by blockaid.

As part of this task we will make the error / warnings messages configurable. The message will depend on the error string returned by blockaid:

### 🖌️ Warning main content:
The content of each warning should follow the logic below:

#### 1️⃣ Case 1: Blockaid's PPOM codename response is _raw_signature_farming_:
**[title]** This is a suspicious request
**[body copy]** If you approve this request, you might lose your assets. 

#### 2️⃣ Case 2: Blockaid's PPOM codename response is _approval_farming_ or _set_approval_for_all_farming_ or _permit_farming_:
**[title]** This is a deceptive request
**[body copy]** If you approve this request, a third party known for scams might take all your assets. 

#### 3️⃣ Case 3: Blockaid's PPOM codename response is _transfer_farming_ or _transfer_from_farming_ or _raw_native_token_transfer_:
**[title]** This is a deceptive request
**[body copy]** If you approve this request, a third party known for scams will take all your assets.  

#### 4️⃣ Case 4: Blockaid's PPOM codename response is _seaport_farming_:
**[title]** This is a deceptive request
**[body copy]** If you approve this request, a third party can steal your assets listed on OpenSea. 

#### 5️⃣ Case 5: Blockaid's PPOM codename response is _blur_farming_:
**[title]** This is a deceptive request
**[body copy]** If you approve this request, a third party can steal your assets listed on Blur. 

#### 6️⃣ Case 6: Blockaid's PPOM codename response is _unfair_trade_ or _other_:
**[title]** This is a deceptive request
**[body copy]** If you approve this request, you might lose your assets. 

### 🖌️ Warning secondary content:
In all cases above, the expandable part of both warnings should contain the following raw data provided by Blockaid's PPOM response: _reason_, _description_, _args_ (this should displayed within the description), _features_ and _score_.

Format:

> _reason_
> - _description_ + _args_
> - _features_
> - This has been rated a risk score of _score_.

Example:
Screenshot 2023-05-05 at 09 16 04

## References
- [Figma file](https://www.figma.com/file/Q69vHaA7Or7Rxz4vdzjVHC/Support-for-tx-security-providers-(MVP)?node-id=5485-37007)
- [Design issue discussion](https://github.com/metamask/metamask-planning/issues/396)
- [PPOM Validation schema and attack types](https://wobbly-nutmeg-8a5.notion.site/PPOM-Validation-Schema-Attack-Types-555e7f6a5b6c4b6caa9ced59561c8667)
- [Intro to component library for an extension engineer](https://www.notion.so/Extension-Engineer-8da34c7887fe4f2182305a5d576d199c?pvs=4)
- [BannerAlert docs](https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-banneralert--default-story)

### PR DESCRIPTION
**Pull request by digiwand**
## Explanation

Fixes: https://github.com/MetaMask/metamask-extension/issues/19257
Blocked by: https://github.com/MetaMask/metamask-extension/pull/19825

### Storybook

zzyJXpp0id

### Tests

Screen Shot 2023-07-17 at 12 35 09 PM

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by jpuri — DISMISSED** [review]
Great work 👍

**Review by blackdevelopa — DISMISSED** [review]

**Review by georgewrmarshall — DISMISSED** [review]
UI LGTM!
- Checked storybook build in PR

File: shared/constants/security-provider.ts [hunk 1]
```diff
@@ -17,7 +17,43 @@ export const SECURITY_PROVIDER_CONFIG: Readonly<SecurityProviderConfig> = {
...
+  /** Malicious signature on a Permit order  */
+  permitFarming = 'permit_farming',
+  /** Direct theft of native assets (ETH/MATIC/AVAX/ etc …)  */
+  rawNativeTokenTransfer = 'raw_native_token_transfer',
+  /** Malicious signature on a Seaport order  */
+  seaportFarming = 'seaport_farming',
+  /** setApprovalForAll for a malicious operator  */
+  setApprovalForAll = 'set_approval_for_all',
+  /** Malicious raw signature from the user   */
+  signatureFarming = 'signature_farming',
```
Comment (segun) [inline shared/constants/security-provider.ts:39]: is this supposed to be based on[ this comment](https://github.com/metamask/metamask-planning/issues/396#issuecomment-1642092127) from Barbara

```suggestion
  rawSignatureFarming = 'raw_signature_farming',
```

File: ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  [BlockaidReason.rawNativeTokenTransfer]: 'blockaidDescriptionTransferFarming',
+  [BlockaidReason.transferFarming]: 'blockaidDescriptionTransferFarming',
+  [BlockaidReason.transferFromFarming]: 'blockaidDescriptionTransferFarming',
+
+  [BlockaidReason.other]: 'blockaidDescriptionMightLoseAssets',
+});
+
+/** List of suspicious reason(s). Other reasons will be deemed as deceptive. */
+const SUSPCIOUS_REASON = [BlockaidReason.seaportFarming];
```
Comment (segun) [inline ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js:40]: The only suspicious reason is rawSignatureFarming from [this comment](https://github.com/metamask/metamask-planning/issues/396#issuecomment-1642092127)
```suggestion
const SUSPCIOUS_REASON = [BlockaidReason.rawSignatureFarming];
```

Comment (digiwand) [reply ui/components/app/security-provider-banner-alert/blockaid-banner-alert/blockaid-banner-alert.js]: thank you! fixed [befa620](https://github.com/MetaMask/metamask-extension/pull/20051/commits/befa620bbaa803f8868d2987a0689a6eb78818b1)

bug happened here: https://github.com/MetaMask/metamask-extension/pull/20051/commits/dcfa47a651b4b785aa7e482ed2a7a33e5f33d39d

**Review by cryptotavares — APPROVED** [review]

**Review by jpuri — APPROVED** [review]

**Pull request merged**